### ISSUE TITLE
Certain icons seem to take a very long time to render in the applications menu

### ISSUE THREAD

**Issue by Lyude** [mate-desktop/mate-panel#75]
When opening the applications menu in the mate-panel and going to office, the icons for Libreoffice seem to take an unusually long time to render, and memory usage seems to go up by around 45 MB after they finally render to the screen.
##

**Comment by elbc56** [comment]
Actually this is still happening with new mate version for submenu where some libreoffice icons are present, if I change the default icon for libreoffice this issue is resolved

**Comment by NiceandGently** [comment]
Does this issue still exists with Mate-1.8/10 ?

**Comment by lifeofguenter** [comment]
Yes, problem still exists with Mate-1.8.1/Debian 8. There must be a smarter way to handle this, like keep a cache and only recheck + rebuild (if source file changed) it if mate-panel starts.

**Comment by ZrN-kAyDqY** [comment]
The response didn't slow for me when changing the theme from the default MATE theme to High Contrast Inverse.  Even when changing the eight Office icons to the largest ones (> 20kb) that were originally pulling from:  /usr/share/icons/Humanity/apps/48

**Comment by Exalm** [comment]
LibreOffice seems to be missing 24x24 and 22x22 icons. It seems that in this case the panel either tries to load the largest ones (256x256 in this case) or scalable icons.

**Comment by AngryPhantom** [comment]
Confirmed here too. LibreOffice misses 24 (and probably 22 icons). Deleting 128, 256, 512 and scalable icons resolves the issue for me. Please, fix that, guys! Thank you in advance.

**Comment by raveit65** [comment]
We are not libreoffice upstream, how should we fix that?
We can't ship trademarked icons 
I suggest to file out a report against libreoffice or your distro..

**Comment by AngryPhantom** [comment]
Why LibreOffice? Why not use the nearest icon set, not the maximum size? 32x32 for example. I don't get it.

**Comment by raveit65** [comment]
Still wondering why it isn't a problem in fedora here, libreoffice don't ship 22/24px icons here too.

**Comment by Exalm** [comment]
Can you check what icon sizes does LO have?
```bash
cd /usr/share/icons/hicolor
find -name "*libreoffice-writer*"
```

**Comment by AngryPhantom** [comment]
> Can you check what icon sizes does LO have?

OpenOffice: 16х16, 32х32 and 48х48 (no problems, MATE chooses 48 and the menu is being populated instantly)
LibreOffice: 16х16, 32х32, 48х48, 128х128, 256х256, 512х512 and scalable (MATE chooses the biggest leading to a notable menu lag

**Comment by Exalm** [comment]
I meant in Fedora, since it doesn't happen for @raveit65

**Comment by raveit65** [comment]
[rave@mother hicolor]$ find -name "*libreoffice-writer*"
./256x256/apps/libreoffice-writer.png
./48x48/apps/libreoffice-writer.png
./32x32/apps/libreoffice-writer.png
./128x128/apps/libreoffice-writer.png
./16x16/apps/libreoffice-writer.png
./scalable/apps/libreoffice-writer.svg
./symbolic/apps/libreoffice-writer-symbolic.svg

**Comment by Exalm** [comment]
You have the same icons as me. Perhaps your computer is just fast enough for you not to notice the lag?

**Comment by raveit65** [comment]
Well, my box isn't slow, but i never noticed this in VM's which are slower than the host.
And i never got a bugzilla report about in the last 3 years.

**Comment by Exalm** [comment]
Ok, then what about icon usage?

I can confirm that the panel uses 50+ MBs of memory after loading the icons: [screencast](https://cloud.githubusercontent.com/assets/1510457/12976230/5dbc1926-d0e4-11e5-9a4a-824f466dbfec.gif)
The lag itself is also very well visible here.

**Comment by raveit65** [comment]
Guys, can someone please test this pull request, this should fix the issue.
https://github.com/mate-desktop/mate-panel/pull/406
Patch link:
https://github.com/XRevan86/mate-panel/commit/13b31fb19e2adc60d69121b09668804cd74ae8a4.diff
Sorry, as i can't see the issue with my box i can't confirm if this fix helps.

**Comment by lukefromdc** [comment]
I just tested this, everything works though with a gtk3 build on Debian Unstable I never saw this issue. Tested ram usage both before and after. possible 2MB reduction in memory usage with the menu open but that's it. Using Libreoffice from Debian Unstable.  Machine in AMD FX 8120 overclocked though, so it would take one hell of a heavy icon set to slow down loading a menu,  but I never even saw this on the netbook. Icon themes full of SVG themes used to bog down Nautilus big time in the circa 2008 era on Pentium 4 class machines though, so no surprise issues like this would come up. I do use SVG's in my gtk theme but they are tiny. My theme might be heavy-and I do remember seeing the old GTK 2 GNOME panel normally use 30+ MB of RAM all the way back go 2008, so I consider more than 30MB but less than 40MB for the panel normal/

**Comment by Exalm** [comment]
GTK3 MATE seems to use 16x16 icons in menu, not sure if it's affected there.
The problem is that LO doesn't ship 24x24 icons, which is a problem, since GTK2 builds of MATE use icons of that size in the menus.

Unfortunately I cannot test this in GTK2 MATE, since I have an ancient version in my distro.

**Comment by raveit65** [comment]
> Unfortunately I cannot test this in GTK2 MATE, since I have an ancient version in my distro.

Sadly nobody post which mate-panel version are used, patch appy clean with 1.12.x

**Comment by Exalm** [comment]
Well, I have 1.8.1 here.

**Comment by Exalm** [comment]
Wait, I have a virtual machine with 1.12.sth. I'll try it now.

**Comment by Exalm** [comment]
Doesn't work. It seems to lag less, but the lag is still noticeable. And the memory usage is actually higher with the patch (68mb instead of 53mb).

**Comment by lukefromdc** [comment]
The smaller menubar icon has been the norm in GTK 3 builds at least back to the 1.9 builds, might be worth looking at. Simplest fix for that though is to make that small icon a fully scaled down version of the larger one, works in my icon theme anyway with a recolored (blue) version of the MATE icon.

**Comment by Exalm** [comment]
Well, this is a GTK2 build, and the smaller icon isn't normal here. :)

**Comment by lukefromdc** [comment]
Using an icon that looks right in both sizes may be a workaround but it wouid work in any gtk version. Alternately fixing it right in the menubar code would probably fix both this issue in the proposed commit and in gtk3 builds.

**Comment by Exalm** [comment]
The problem is that the menu icon size seems to be the only thing the patch does. :/

**Comment by raveit65** [comment]
Ok, looks like we can't do anything from our side.
I really suggest to file out a report against libreoffice at your distro and ask them to add missing icons, i guess 1.8.1 comes from deprecated debian stable.
That's the only way to fix that.

General:
1.12. is current release, and we will never fix older versions (we are a small team).
If a distro don't update than downstream maintainer is responsable to backport fixes.

**Comment by XRevan86** [comment]
> The problem is that the menu icon size seems to be the only thing the patch does. :/

@Exalm, not anymore :-).

**Comment by raveit65** [comment]
Does this issue still exists?
